src/crypto/crypto.h

```diff
@@ -152,6 +152,20 @@ namespace crypto {
     return res;
   }
 
+  /* Generate an index between 0 and a given value with fixed probability
+   */
+  inline size_t random_index_with_fixed_probability(size_t max_index)
+  {
+    //divide by zero workaround
+    if(!max_index)
+      return 0;
+
+    size_t x = crypto::rand<size_t>()%(max_index+1);
+    size_t res = (x*x*x)/(max_index*max_index); //parabola \/
```

Comment:
Integer overflow on most of the range, as a side note. But since it's just moving around, ok.